lib/active_record_shards/default_slave_patches.rb

```diff
@@ -100,7 +100,10 @@ def on_slave_unless_tx
 
     module ActiveRelationPatches
       def self.included(base)
-        [:calculate, :exists?, :pluck, :find_with_associations].each do |m|
+        [
+          :calculate, :exists?, :pluck,
+          ActiveRecord::VERSION::STRING >= '5.2.0' ? :load : :find_with_associations
```

Comment:
The commit message for rails/rails@412db710dfa6ed84654068576b1841966d7f89b2 is “Use `apply_join_dependency` instead of meaningless named `find_with_associations`” – I wonder why `:load` was chosen here instead of `:apply_join_dependency`.

Reply:
`apply_join_dependency` is an internal method. We have to do the change because internal methods tend to be changed with no backward compatibility. Also `apply_join_dependency` is used in many places - when we hack it, we are not sure which public methods are actually get hacked.

Reply:
That makes sense. What about `:load`, that sounds like a generic method too?

Does `:load` exist in older versions of Rails, so we could just patch that method instead of `find_with_associations`?